### PR TITLE
update kubernetes-client version to 5.12.1

### PR DESCRIPTION
**Pull request by kyoppii13**
Updated version of k8s client to 5.12.1.

The reason is as follows.
* I want to use the backoff parameter (kubernetes.request.retry.backoffLimit).
* Since the supported version of k8s cluster was old.

https://github.com/fabric8io/kubernetes-client

[kubernetes-client migration document fromt 4.x to 5.x](https://github.com/fabric8io/kubernetes-client/blob/master/doc/MIGRATION-v5.md)

### CODE REVIEW

**Comment by yoyama** [comment]
LGTM. But it might be better to add some tests.

**Comment by kyoppii13** [comment]
@yoyama 
I resolve conflicts.
Please review.

File: digdag-standards/src/test/java/io/digdag/standards/command/kubernetes/DefaultKubernetesClientTest.java [hunk 1]
```diff
@@ -45,8 +47,8 @@
 import static io.digdag.core.workflow.OperatorTestingUtils.newTaskRequest;
 import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
+import static org.mockito.Matchers.*;
```
Comment (yoyama) [inline digdag-standards/src/test/java/io/digdag/standards/command/kubernetes/DefaultKubernetesClientTest.java:50]: Please not use `*`

**Comment by yoyama** [comment]
@kyoppii13 I left comments on minor things. Please check it.

**Review by yoyama — APPROVED** [review]
LGTM 👍 
Thank you for your contribution.

**Pull request merged**